### PR TITLE
feat: cache dependencies between workflows

### LINKED ISSUE #93: Cache dependencies between workflows
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows

### PR DESCRIPTION
**Pull request by km1chno**
Resolves #93 

## Changes

- add `Make sure you generated lock file by installing project dependencies.` message to error when lock file is not detected
- find node version file `.nvmrc`/`.node-version`/`package.json` and pass its path to workflows to use in `setup node` action (create `.nvmrc.` with node `v20.17.0` if no file found)
- add 'setup node' action in all workflows to cache yarn/npm dependencies and fix node version
- change installation scripts to `npm ci` and `yarn install --immutable` for consistent runs
- fix some indentation problems in workflow templates

### CODE REVIEW

File: src/templates/build-release/build-release-android.ejf [hunk 1]
```diff
@@ -49,6 +49,12 @@ jobs:
       - name: 🏗 Checkout repository
         uses: actions/checkout@v4
 
+      - name: 🌿 Setup Node 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
```
Comment (maciekstosio) [inline src/templates/build-release/build-release-android.ejf:55]: I wonder if we shouldn't guess the node version somehow instead of hardcoding it

Comment (km1chno) [reply src/templates/build-release/build-release-android.ejf]: added dynamic determination of node version in order respecting https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file:
1. `.nvmrc` file in **package root**
2. `.node-version` file in **package root** 
3. `engines.node` or `volta.node` field in `package.json` in **package root** 
4. 1. `.nvmrc` file in **repo root**
5. `.node-version` file in **repo root**
6. `engines.node` or `volta.node` field in `package.json` in **repo root**

and if 1-6 fail, create `.nvmrc` with `v20.17.0` in **package root**

then we use the path to node version file in `Setup node` step with field `node-version-file`

File: src/extensions/projectConfig.ts [hunk 2]
```diff
@@ -29,6 +31,43 @@ module.exports = (toolbox: CycliToolbox) => {
...
+    // If still not found, create .nvmrc in package root with default node version (v20.17.0).
+    const file =
+      nodeVersionFileInDirectory(packageRoot) ??
+      nodeVersionFileInDirectory(repoRoot)
+
+    if (!file) {
+      filesystem.write(join(packageRoot, '.nvmrc'), DEFAULT_NODE_VERSION)
+
+      toolbox.interactive.warning(
+        'No node version file found. Created .nvmrc with default node version (v20.17.0).'
```
Comment (maciekstosio) [inline src/extensions/projectConfig.ts:62]: pls change hardcode for the variable + I'm thinking about adding this to what's next too - I think setting node version might be quiet a big change

Comment (km1chno) [reply src/extensions/projectConfig.ts]: I agree, changed hardcoded variable to constant and added info to further actions box in 7f56f1810e19ed928420572080d4411263f36086

**Review by maciekstosio — APPROVED** [review]
LGTM 🚀

**Pull request merged**